lib/granite/context_proxy/data.rb

```diff
@@ -0,0 +1,20 @@
+module Granite
+  module ContextProxy
+    # Contains all the arbitrary data that is passed to BA with `using`
+    class Data
+      attr_reader :performer
+
+      def self.wrap(data)
+        if data.is_a?(self)
+          data
+        else
+          new(**data || {})
```

Comment:
If data is a `Hash` then we don't need ` || {}`
If data is `nil` then I would expect `**data` to crash but it does not. It seems to be working as `new(**(data || {}))` but I can't repro in my console. Maybe older ruby or something...

I think it should crash with `nil`, what do you think?

Reply:
I used this example script and checked with ruby 2.6, 2.7, 3.0 and 3.1:
```ruby
def print_data(**data)
  p data
end
print_data(**nil || {})
```

All of them correctly printed `{}`, so I assume all of them interpret it as `new(**(data || {}))`. The reason I'm allowing nil is the following situation:

```ruby
class ParentAction
  def child_action
    ChildAction.using(ctx).new
  end
end

ParentAction.new.child_action
```

The last line would crash if `nil` is not allowed since `parent_action.ctx` is `nil`.
We could move `nil` handling to `proxy_context` method though and return "empty" context there, however that means if you need to override your custom context class in 2 places: `Granite::ContextProxy::ClassMethods#using` and `Granite::ContextProxy::ClassMethods#proxy_context`.